lib/compiler/src/engine/artifact.rs

```diff
@@ -572,6 +570,10 @@ impl Artifact {
             )?;
         let mut obj = get_object_for_target(target_triple).map_err(to_compile_error)?;
 
+        let object_name = format!(
+            "WASMER_{}_METADATA",
```

Comment:
I think `WASMER_METADATA_{prefix}` is a better one

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3430/commits/2c9ad00573dfb76c986a52a87e29f83bd8a7c384